assets/js/modules/analytics/setup.js

```diff
@@ -87,11 +85,39 @@ class AnalyticsSetup extends Component {
 	async componentDidMount() {
 		this._isMounted = true;
 
-		this.iframeLoad();
+		await this.getAccounts().then( async( newState ) => {
 
-		await this.getAccounts();
+			// If tag hasn't been detected in wp_head, look for existing tag in the html.
+			if ( ! newState.existingTag && ! newState.errorCode ) {
+				const { onSettingsPage } = this.props;
+				const { useSnippet } = this.state;
+				const existingTag = await getExistingTag( 'analytics' );
 
-		this.getExistingTag();
+				if ( existingTag && 'no tag' !== existingTag ) {
```

Comment:
See other comments, shouldn't we instead return an empty string and check `existingTag.length`?

Reply:
addressed here https://github.com/google/site-kit-wp/pull/82/commits/a005b3d9e39a7d5f94a932a836d24132e57ce6ed